Source/JavaScriptCore/interpreter/ProtoCallFrameInlines.h

```diff
@@ -37,13 +37,10 @@ inline void ProtoCallFrame::init(CodeBlock* codeBlock, JSGlobalObject* globalObj
     this->setCallee(callee);
     this->setGlobalObject(globalObject);
     this->setArgumentCountIncludingThis(argCountIncludingThis);
+    size_t paddedArgsCount = argCountIncludingThis;
     if (codeBlock && static_cast<unsigned>(argCountIncludingThis) < codeBlock->numParameters())
-        this->hasArityMismatch = true;
-    else
-        this->hasArityMismatch = false;
-
-    // Round up argCountIncludingThis to keep the stack frame size aligned.
-    size_t paddedArgsCount = roundArgumentCountToAlignFrame(argCountIncludingThis);
+        paddedArgsCount = codeBlock->numParameters();
+    paddedArgsCount = roundArgumentCountToAlignFrame(paddedArgsCount);
```

Comment:
What is the case where CodeBlock is null? In that case, why is it OK to assume that arity matches?

On the surface, this appears to be a correctness regression. The previous code calculated 'hasArityMismatch = false', but then dropped that value on the floor -- two bugs cancelling each other out. In this patch, when codeBlock is null, we jump directly to the ArityCheckNotRequired entry point -- potentially causing the callee to dereference uninitialized argument memory.

Reply:
When CodeBlock is null, we're looking at a Wasm or Native function.  In those cases, their normal and arity check entries both point to the same entry anyway.  They do not have an arity check version.  The wasm and native function also does not access arguments beyond the count is set in `CallFrame::argumentIncludingThis`.  This is already the case in present code.

`hasArityMismatch` is unused code.  No one checks it anymore.